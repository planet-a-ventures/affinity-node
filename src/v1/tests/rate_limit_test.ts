import { assertSnapshot } from '@std/testing/snapshot.ts'
import { afterEach, beforeEach, describe, it } from '@std/testing/bdd.ts'

import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'
import Affinity from '../index.ts'
import { getRawFixture } from './get_raw_fixture.ts'
import { isLiveRun, apiKey } from './env.ts'

describe('rate_limit', () => {
    let mock: MockAdapter
    let affinity: Affinity

    beforeEach(() => {
        if (!isLiveRun()) {
            mock = new MockAdapter(axios)
        }
        affinity = new Affinity(apiKey() || 'api_key')
    })
    afterEach(() => {
        mock?.reset()
    })

    it('can be called', async (t) => {
        mock?.onGet('/rate-limit').reply(
            200,
            await getRawFixture('rate_limit/rate_limit.raw.response.json'),
        )
        const res = await affinity.rateLimit()
        await assertSnapshot(t, res)
    })
})
