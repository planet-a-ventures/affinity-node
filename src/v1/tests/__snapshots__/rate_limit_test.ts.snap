export const snapshot = {};

snapshot[`rate_limit > can be called 1`] = `
{
  rate: {
    api_key_per_minute: {
      limit: 900,
      remaining: 878,
      reset: 10,
      used: 22,
    },
    org_monthly: {
      limit: Infinity,
      remaining: Infinity,
      reset: 2124845,
      used: 7,
    },
  },
}
`;
